File: tournaments/create2DArray/create2DArray.cs
int[][] create2DArray(int[] lengths) {
    var n = lengths.Length;
    var result = new int[n][];
    
    for (var i = 0; n > i; ++i) {
        result[i] = new int[lengths[i]];
        for (int j = 0; lengths[i] > j; ++j) {
            result[i][j] = j;
        }
    }   
    
    return result;
}

File: tournaments/create2DArray/create2DArray.cpp
std::vector<std::vector<int>> create2DArray(std::vector<int> lengths) {

  std::vector<std::vector<int>> result;

  for (int i = 0; i < lengths.size(); i++) {
    std::vector<int> line;
    for (int j = 0; j < lengths[i]; j++) {
      line.push_back(j);
    }
    result.push_back(line);
  }

  return result;
}
